pkg/devconsole/handler.go

```diff
@@ -0,0 +1,92 @@
+package devconsole
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/openshift/console/pkg/auth"
+	"github.com/openshift/console/pkg/devconsole/artifacthub"
+	tektonresults "github.com/openshift/console/pkg/devconsole/tekton-results"
+	"github.com/openshift/console/pkg/devconsole/webhooks"
+	"github.com/openshift/console/pkg/serverutils"
+	"k8s.io/client-go/dynamic"
+)
+
+type handlerFunc func(r *http.Request, user *auth.User, dynamicClient *dynamic.DynamicClient) (interface{}, error)
+
+func handleRequest(w http.ResponseWriter, r *http.Request, user *auth.User, dynamicClient *dynamic.DynamicClient, handler handlerFunc) {
+	response, err := handler(r, user, dynamicClient)
+	if err != nil {
+		serverutils.SendResponse(w, http.StatusInternalServerError, serverutils.ApiError{Err: err.Error()})
+		return
+	}
+	serverutils.SendResponse(w, http.StatusOK, response)
+}
+
+func Handler(user *auth.User, w http.ResponseWriter, r *http.Request, dynamicClient *dynamic.DynamicClient) {
+	path := strings.Split(strings.Trim(r.URL.Path, "/"), "/")
+	if len(path) != 2 {
+		serverutils.SendResponse(w, http.StatusNotFound, serverutils.ApiError{Err: "Invalid URL"})
+		return
+	}
+
+	// Define routes
+	routes := map[string]map[string]handlerFunc{
```

Comment:
It feels like we should register these as separate handlers in server.go instead of handling this as a map here